__tests__/users.ts

```diff
@@ -320,19 +320,41 @@ describe('query userStreaks', () => {
     }
   }`;
 
+  const QUERY_BY_USER_ID = `query UserStreak($id: ID) {
+    userStreak(id: $id) {
+      max
+      total
+      current
+      lastViewAt
+    }
+  }`;
+
   afterEach(() => {
     jest.useRealTimers();
   });
 
-  it('should not allow unauthenticated users', () =>
-    testQueryErrorCode(client, { query: QUERY }, 'UNAUTHENTICATED'));
+  it('should throw error when no user id provided and no user is authenticated', async () => {
+    await testQueryErrorCode(
+      client,
+      { query: QUERY },
+      'GRAPHQL_VALIDATION_FAILED',
+      'User is missing!',
+    );
+  });
 
   it('should return the user streaks', async () => {
     loggedUser = '1';
     const res = await client.query(QUERY);
     expect(res.errors).toBeFalsy();
   });
 
+  it('should return the user streaks based on user id', async () => {
+    const resById = await client.query(QUERY_BY_USER_ID, {
+      variables: { id: '2' },
+    });
+    expect(resById.errors).toBeFalsy();
```

Comment:
You probably also want to ensure the actual data here on top of no errors.

Reply:
That will always return the default data, added that test. There is also a test doing just that on https://github.com/dailydotdev/daily-api/blob/cca8714abe73b7efcaf653684b9225e6095f4a74/__tests__/users.ts#L428.

Reply:
Perfect, just extra failsafe if someone changes defaults by accident :)